src/bulls_and_cows_lib/board.cpp

```diff
@@ -1,8 +1,76 @@
 
 #include "board.hpp"
+#include "random.hpp"
+#include <algorithm>
 
 namespace bulls_and_cows {
+    using namespace std;
+    Board create_board(const GameOptions& game_options)
+    {
+        Board board;
+        std::string code{};
+        for (unsigned i = 0; i < game_options.number_of_characters_per_code; i++)
+        {
+            code += generate_random_character(game_options.minimum_allowed_character,
+                                              game_options.maximum_allowed_character);
+        }
+        board.secret_code.value = code;
+        return board;
+    }
 
-    // TODO: define the body of the functions declared in board.cpp
+    bool validate_attempt(const GameOptions& game_options, const Code& attempt)
+    {
+        if (attempt.value.length() != game_options.number_of_characters_per_code)
+        {
+            return false;
+        }
 
-} // namespace bulls_and_cows
+        for (unsigned i = 0; i < game_options.number_of_characters_per_code; i++)
```

Comment:
vois-tu comment faire une boucle range-for (https://en.cppreference.com/w/cpp/language/range-for) au lieu d'une boucle for ?